### PR TITLE
UI: Show link to release notes in Help menu

### PR DESCRIPTION
**Pull request by WizardCM**
### Changes from the original PR #2544

* No longer calls an API, simply shows the menu item if the version is considered 'valid'

### Original Description

Dynamically shows a Release Notes link if the `OBS_VERSION` is valid.

image

### Motivation and Context

Currently, users are only exposed to the release notes once - before they update. This provides a quick and easy link once updated that doesn't require digging.

### How Has This Been Tested?

* Build with a valid version CMake variable. See the menu item.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by WizardCM** [comment]
Per internal discussion, I've removed the network calls entirely. Still need to do actual semver validation in the `aboutToShow()` call, and then this'll be ready.

**Comment by cg2121** [comment]
I would put this in the About dialog. This is how Firefox does it.
2022-05-17 03_50_35-About Mozilla Firefox

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -7427,6 +7427,13 @@ void OBSBasic::on_actionDiscord_triggered()
 	QDesktopServices::openUrl(url);
 }
 
+void OBSBasic::on_actionReleaseNotes_triggered()
+{
+	QString addr("https://github.com/obsproject/obs-studio/releases");
+	QUrl url(QString("%1/%2").arg(addr, OBS_VERSION), QUrl::TolerantMode);
```
Comment (derrod) [inline UI/window-basic-main.cpp:7433]: With #8930 merged this should use `obs_get_version_string()` instead. Or alternatively `OBSApp::GetVersionString()´ which will also work if `OBS_VERSION` is not defined.

File: UI/window-basic-main.cpp [hunk 2]
```diff
@@ -9302,6 +9309,16 @@ void OBSBasic::on_stats_triggered()
...
 }
 
+void OBSBasic::on_menuBasic_MainMenu_Help_aboutToShow()
+{
+#ifdef HAVE_OBSCONFIG_H
+	QString ver;
+	ver += OBS_VERSION;
+	if (!ver.isEmpty())
+		ui->actionReleaseNotes->setVisible(true);
+#endif
```
Comment (derrod) [inline UI/window-basic-main.cpp:9319]: Constructing a string here is probably overkill. As long as the header is present the version should be nonempty.
```suggestion
#ifdef HAVE_OBSCONFIG_H
	ui->actionReleaseNotes->setVisible(true);
#endif
```

File: UI/window-basic-main.hpp [hunk 3]
```diff
@@ -348,6 +348,7 @@ class OBSBasic : public OBSMainWindow {
 	bool QueryRemoveSource(obs_source_t *source);
 
 	void TimedCheckForUpdates();
+	bool releaseNotesChecked = false;
```
Comment (derrod) [inline UI/window-basic-main.hpp:351]: This seems to be no longer used.

**Review by derrod — CHANGES_REQUESTED** [review]
I was thinking about doing something similar myself recently but then discovered this PR exists. It needs a few updates but should otherwise be fine.

File: UI/window-basic-main.hpp (hunk 3 above)
Comment (WizardCM) [reply UI/window-basic-main.hpp]: You're right, removed.

File: UI/window-basic-main.cpp (hunk 2 above)
Comment (WizardCM) [reply UI/window-basic-main.cpp]: Good point, fixed.

File: UI/window-basic-main.cpp (hunk 1 above)
Comment (WizardCM) [reply UI/window-basic-main.cpp]: Ended up using `obs_get_version_string`, seems to work fine.

**Comment by derrod** [comment]
On second thought, is `OBSBasic::on_menuBasic_MainMenu_Help_aboutToShow()` actually necessary? Can you not modify the visibiilty in the constructor of `OBSBasic`?

**Comment by WizardCM** [comment]
@derrod Good catch, fixed.

**Pull request merged**